src/components/SearchByCity.js

```diff
@@ -0,0 +1,15 @@
+function SearchByCity() {
+  return `
+    <input
+        type="text"
+        value="${window.dataStore.currentCity}"
+        onchange="
+            window.dataStore.currentCity = this.value; 
+            window.dataStore.error = null; 
+            window.dataStore.isDataLoading = true; 
```

Comment:
Looks like an antipattern :) Our code will be duplicated by students, who mostly will copy-paste it.

Please, move this logic into the local function inside the component.